Add exception handler to scan action.

Cypresslin opened the pull request:
Ask the interface to restart discovery if it's already in scanning.
- Not a very good solution, it might also fail to stop / start discovery in that if, and making runaway exceptions.

Cypresslin committed:
Add exception handler to scan action.

Ask the interface to restart discovery if it's already in scanning.

* Not a very good solution, it might also fail to stop / start discovery in that if, and making runaway exceptions.

Cypresslin commented:
Or maybe we should not try to restart the scanning and just let it fail? (not to put Start/StopDiscovery in the except section.)

kissiel commented:
It could be easier if we just .StopDiscovery first, in a try/except, and then just .StartDisco(). That way starting disco will not nest try/excepts and we can better differentiate between exceptions.

kissiel commented on bt_helper.py:
```diff
@@ -80,7 +80,16 @@ def scan(self):
-            dbus.Interface(adapter, ADAPTER_IFACE).StartDiscovery()
+            try:
+                dbus.Interface(adapter, ADAPTER_IFACE).StartDiscovery()
+            except Exception as exc:
```
s/Exception/DBusException

Cypresslin replied:
If I use DBusException here, I will need to use dbus.exceptions.DBusException instead, or define it with something else, any suggestions?

And I think maybe Exception class can allow us to catch something other than DBusExceptions?

kissiel replied:
I guess the things we should catch here are DBusExceptions, as those are the ones we want to handle. If anything else is raised we should IMHO let it through, as this signifies something weird/bad/important.

Cypresslin replied:
OK, this makes a perfect sense!

Cypresslin commented:
Ooooh, that's a good idea, I'll fix that. Thanks

Cypresslin committed:
Scan exception improved, stop it first with try then start it later.

Ignore exceptions from DBusException, allow other exceptions to crash.
Just try to stop discovery, don't react with the DBusException as the
error message 'No discovery started' could be changed.

Cypresslin committed:
Remove exc for exception handling in scan function.

We are not using exc in the exception handling function for scan, remove it.

kissiel commented:
Beautiful, +1.
